packages/rs-drive/src/drive/cache.rs

```diff
@@ -20,4 +23,6 @@ pub struct DriveCache {
     pub protocol_versions_counter: parking_lot::RwLock<ProtocolVersionsCache>,
     /// Versioned system data contracts
     pub system_data_contracts: SystemDataContracts,
+    /// Cached Epoch changed FeeVersion
+    pub cached_fee_version: parking_lot::RwLock<BTreeMap<EpochIndex, &'static FeeVersion>>,
```

Comment:
You better use thread-safe map-based structures than ReadWrite lock. It will be much more efficient. DashMap for example.

Reply:
@shumkov Unfortunately, `DashMap` doesn't provide order of keys. By searching on the net, it seems that neither `type IntMap<K, V> = std::collections::HashMap<K, V, BuildNoHashHasher<K>>` guarantees order of keys. That is why I chose `BTreeMap`.

Regarding performance, this map will be updated only when `FeeVersion` is changed. At worst case, `FeeVersion` is changed on every epoch (=40/year) so it is really not a frequent write pattern.
The read is much more frequent but again, the map will hold very few values. O(log n) with up to 40 key-value pair per year isn't scary.

Reply:
All good